src/health_checker.cc

```diff
@@ -39,12 +39,31 @@ void HealthChecker::SetUnhealthy(const std::string& component) {
 
 bool HealthChecker::IsHealthy() const {
   std::lock_guard<std::mutex> lock(mutex_);
-  return unhealthy_components_.empty();
+  if (!unhealthy_components_.empty()) {
+    return false;
+  }
+  for (auto& c : component_callbacks_) {
+    if (!c.second()) {
+      return false;
+    }
+  }
+  return true;
 }
 
 void HealthChecker::CleanupForTest() {
   boost::filesystem::remove_all(boost::filesystem::path(
       config_.HealthCheckFile()).parent_path());
 }
 
+void HealthChecker::RegisterCallback(const std::string& component,
```

Comment:
[Optional] I'd call this `AddComponent` or `RegisterComponent` instead... You're adding a pair of <component_name, callback>.

Reply:
Done.